active/RowColumnAPIChange/rowcolumnapichange.md

```diff
@@ -0,0 +1,168 @@
+1. Create a Column or RowDefinition object with the blank constructor.
+2. Define a GridLength object with the desired value for the Height or Width property of the Row/ColumnDefinition.
+3. Assign the GridLength object to the Height or Width property of the Row/ColumnDefinition. 
+
+This spec sets out to change that process, and therefore sets out to change the [RowDefinition](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.RowDefinition) and [ColumnDefinition](https://docs.microsoft.com/uwp/api/windows.ui.xaml.controls.columndefinition) APIs.
+
+This new feature will allow developers to define their column widths and row heights within the column and row definitions themselves, not only making the code-behind easier to write, but making it cleaner and more understandable to read as well. 
+
+# Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+
+ColumnDefinition and RowDefinition constructors will be overloaded to take in one or two arguments. The first overloaded constructor for `ColumnDefinition()` will take only a double-type argument called Width (Height for `RowDefinition()`). The second overloaded constructor for ColumnDefinition() will take a double-type argument Width (Height for RowDefinition()) and a [GridUnitType](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.gridlength.gridunittype) argument.
+
+These overloaded constructors take the same arguments as the [GridLength](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.GridLength) constructor does - a double and a GridUnitType. In turn, feeding these same arguments to a RowDefinition or ColumnDefinition constructor will create a GridLength object and assign it to the ColumnDefinition's Width property or the RowDefinition's Height property behind the scenes. 
+
+# Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+Current syntax to define a Grid with one column in code-behind:
+```csharp
+var grid = new Grid();
+var column = new ColumnDefinition();
+column.Width = new GridLength(1.0, GridUnitType.Star);
+grid.ColumnDefinitions.Add(column);
```

Comment:
This comparison seems a little unfair. These 4 lines could be written as:
```cs
var grid = new Grid();
grid.ColumnDefinitions.Add(new ColumnDefinition() { Width = new GridLength(1.0, GridUnitType.Star) });
```
That makes it almost identical to the new syntax (just not using the new proposed constructor)

Reply:
If one uses C#, you can write that. In C++ you don't have that C# syntactic sugar.

Reply:
Good points. It's true that you don't get the same kind of syntactic sugar in C++, but also I was mainly writing the sample out in this way so it makes sense with the steps that I listed above:

> 1. Create a Column or RowDefinition object with the blank constructor.
> 2. Define a GridLength object with the desired value for the Height or Width property of the Row/ColumnDefinition.
> 3. Assign the GridLength object to the Height or Width property of the Row/ColumnDefinition.